crypto/sessionkey.go

```diff
@@ -171,65 +172,85 @@ func (sk *SessionKey) EncryptWithCompression(message *PlainMessage) ([]byte, err
 
 func encryptWithSessionKey(message *PlainMessage, sk *SessionKey, signEntity *openpgp.Entity, config *packet.Config) ([]byte, error) {
 	var encBuf = new(bytes.Buffer)
-	var encryptWriter, signWriter io.WriteCloser
 
-	encryptWriter, err := packet.SerializeSymmetricallyEncrypted(encBuf, config.Cipher(), sk.Key, config)
+	encryptWriter, signWriter, err := encryptStreamWithSessionKey(
+		message.IsBinary(),
+		message.Filename,
+		message.Time,
+		encBuf,
+		sk,
+		signEntity,
+		config,
+	)
 	if err != nil {
-		return nil, errors.Wrap(err, "gopenpgp: unable to encrypt")
+		return nil, err
+	}
+	if signEntity != nil {
+		_, err = signWriter.Write(message.GetBinary())
+		if err != nil {
+			return nil, errors.Wrap(err, "gopenpgp: error in writing signed message")
+		}
+		err = signWriter.Close()
+		if err != nil {
+			return nil, errors.Wrap(err, "gopenpgp: error in closing signing writer")
+		}
+	} else {
+		_, err = encryptWriter.Write(message.GetBinary())
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "gopenpgp: error in writing message")
+	}
+	err = encryptWriter.Close()
+	if err != nil {
+		return nil, errors.Wrap(err, "gopenpgp: error in closing encryption writer")
+	}
+	return encBuf.Bytes(), nil
+}
+
+func encryptStreamWithSessionKey(
+	isBinary bool,
+	filename string,
+	modTime uint32,
```

Comment:
Generally we use int64 for the top level API, that internally is cast to uint32 in x/crypto. I'd suggest doing the same here

Reply:
That's not the top level api right ? That's an internal method

Reply:
Yeah, true. Since we use it twice and cast it once I'd still prefer to use here int64 :)

Reply:
The problem is that `encryptStream` is also called here: https://github.com/ProtonMail/gopenpgp/blob/feat/streaming-api/crypto/sessionkey.go#L176
and message.Time is already a uint32, so if I put int64 then I have to cast it twice